src/main/java/org/elasticsearch/SecureSM.java

```diff
@@ -170,28 +162,13 @@ protected void checkThreadGroupAccess(ThreadGroup g) {
   // exit permission logic
   @Override
   public void checkExit(int status) {
-    if (allowTestExit) {
-      checkTestExit(status);
-    } else {
-      throw new SecurityException("exit(" + status + ") not allowed by system policy");
-    }
+    checkExit(status, packagesThatCanExit);
   }
-
-  static final String TEST_RUNNER_PACKAGES[] = {
-    // surefire test runner
-    "org.apache.maven.surefire.booter.",
-    // junit4 test runner
-    "com.carrotsearch.ant.tasks.junit4.",
-    // eclipse test runner
-    "org.eclipse.jdt.internal.junit.runner.",
-    // intellij test runner
-    "com.intellij.rt.execution.junit."
-  };
```

Comment:
I think that it would be odd that the default constructor would allow test packages to exit instead of no packages to exit? I think we can expose these test packages as a convenience field though?